microsvcs/admin/forum/app/model/forum_model.js

```diff
@@ -10,11 +10,15 @@ const collection = database.collection(process.env.FORUM_COLLECTION);
  * No parameters expected.
  * @return An Array of all documents contained in the collection.
  */
-exports.getAllDocuments = async () => {
+exports.getAllPosts = async () => {
     try {
-        let result = await collection.find().toArray();
+        const query = { flag: { $not: { $eq: "D" } } };
```

Comment:
this is related to the GET all and DELETE one methods for the forum posts.

the DELETE method rather than removing the database record entirely, it sets a new status flag. so its more like an archival system 

```
{
  "_id": { ... },
  "created_by": "sammy@email.com",
  "timestamp": "2024-06-05T09:21:52.339Z",
  "post_content": { ... },
  "lastModified": { ... },
  "status": {
    "flag": "D",
    "raised_by": "cheryl@email.com"
  }
}
```
that being said, its abit inconsistent with the comment's DELETE method, which removes the comment from the record entirely

relating back to that query, that query filters and doesn't return posts flagged for deletion (for user side viewing), though that being said i didn't write an endpoint which can fetch the "D" flagged posts (endpoint for consideration?)

Reply:
Interesting that you chose to flag it instead of deleting it but we can go with this anyways!

It is a bit inconsistent but we can let it go for now

We probably dont need the endpoint for "D" flagged posts unless you have extra time for admin dashboard etc.